test/ngRoute/routeParamsSpec.js

```diff
@@ -77,5 +77,24 @@ describe('$routeParams', function() {
     });
   });
 
+  it('should correctly extract path params containing hashes and/or question marks', function() {
```

Comment:
Should we test for `$location.path('/foo/bar?baz#qux');`?

And is it intended that the "params" have no values, i.e `?baz=val`?

Should we test if multiple params are correctly extracted?

Reply:
Sure. This is not the primary focus of this test, but I'm never one to shy away from adding more testcases 😛